src/core/device.c

```diff
@@ -152,21 +151,94 @@ static int device_coldplug(Unit *u) {
         return 0;
 }
 
+static const struct {
+        DeviceFound flag;
+        const char *name;
+} device_found_map[] = {
+        [DEVICE_NOT_FOUND] = { DEVICE_NOT_FOUND,     "not-found"     },
```

Comment:
This is essentially a noop, because we know that `DEVICE_NOT_FOUND == 0`, and arrays are also indexed from zero, but with the indentation it's quite misleading and looks like a nested array. Please just drop the `[DEVICE_NOT_FOUND] = ` part.